### ISSUE TITLE
Integrate Wiki Page as help dialog

### ISSUE THREAD

**Issue by kiozen** [Maproom/qmapshack#108]
**Is your feature request related to a problem? Please describe.**
It would be nice to have the Wiki pages as offline help built into the app. 

**Describe the solution you'd like**
Let's try to use the Qt Help framework. 

**Describe alternatives you've considered**
There are two ways. Either use Qt Assistent or write our own help dialog with Qt Help. I favour the last.

**Additional context**
n/a

**Comment by wthaem** [comment]
After looking at some examples I, too, would prefer a help dialog within QMS.

I attach 3 configuration files for a QMS/QMT help package based on Qt which reveal the internals of the package.

For referencing QMS Wiki files in the Qt help use the following syntax

~~~
qthelp:/qms/doc/Home.html
~~~

`Home` can be replaced with the name of an existing QMS Wiki page (e.g. `DocGisDatabaseLostFound`. For the QMT part replace `qms` with `qmt`.

[QMTHelp.zip](https://github.com/Maproom/qmapshack/files/4234622/QMTHelp.zip)

**Comment by kiozen** [comment]
I made a quick, hacky test with the Qt Help Engine:

https://github.com/kiozen/qmapshack/tree/QMS-108

On Linux you have to do:

`make install`

to place the files in the correct path. For Windows I have no clue. Best you start with logging and open the `new help`. The log will tell you where it expects the help files.

@wthaem btw the files you gave me seem to be a bit weird. Somehow QMS and QMT docs got mixed up and the only files working are QMSHelp.* showing the help for QMT. But anyway that's ok for testing.

**Comment by wthaem** [comment]
@kiozen: I can confirm your observation. Please open QMSHelp.qhc with the assistant as described earlier to see the difference. assistant shows what is defined in the help configuration files. Thus, my feeling is, that some initialization of the help engine within QMS is not yet completely correct or incomplete but I don't know, how to do better. And I would add more to the help browser to make it more comfortable: the content, index and search tabs would be nice. At the very end, I hope to see something similar to what is displayed in assistant.

What disturbs me is a file `QMTHelp.qhc` (appearing in `CMakeLists.txt`, but only for UNIX). Where does this file come from? How is it used? In assistant no content is shown for this file. In my opinion it should be dropped. The file `QMSHelp.qhc` consists of the 2 packages `QMSHelp.qch` and `QMTHelp.qch`. Therefore, the top level of the content tree shows 2 entries: `QMS manual` and `QMT manual`. In the index tab the QMS and the QMT index parts are merged.

Maybe. this link gives some clarification:

~~~
https://doc-snapshots.qt.io/qt5-5.12/qthelp-contextsensitivehelp-example.html
~~~


The required location of the help files is in my (Windows) case:

~~~
some_root_directory\msvc_64\Files\doc\HTML\QMSHelp.q*
~~~

When using the Qt help (at least with the assistant) a cache folder `some_root_directory\msvc_64\Files\doc\HTML\cache` is created and some files are written there (during runtime). If `some_root_directory` is set by the Windows installation routine to `Program Files` this may cause trouble. But there is a Qt help installation parameter that allows to move the cache directory elsewhere.


BTW: I updated Qt5 to version 5.14.0 and can use it without any problems when compiling and running the dev branch of QMS and QMT (there is quite a lot of additional warnings, of course). But your new branch didn't compile:

~~~
3>CHelp.cpp
3>D:\QtProjects\QMS\QMS_Kiozen\src\qmapshack\CHelp.cpp(45): error C2664: "QMetaObject::Connection QObject::connect(const QObject *,const char *,const char *,Qt::ConnectionType) const" : Konvertierung von Argument 2 von "void (__cdecl QHelpContentWidget::* )(const QUrl &)" in "const char *" nicht möglich
3>D:\QtProjects\QMS\QMS_Kiozen\src\qmapshack\CHelp.cpp(45): note: Es gibt keinen Kontext, in dem diese Konvertierung möglich ist
3>D:\QtProjects\QMS\QMS_Kiozen\src\qmapshack\CHelp.cpp(46): error C2664: "QMetaObject::Connection QObject::connect(const QObject *,const char *,const char *,Qt::ConnectionType) const" : Konvertierung von Argument 2 von "void (__cdecl QHelpIndexWidget::* )(const QUrl &,const QString &)" in "const char *" nicht möglich
3>D:\QtProjects\QMS\QMS_Kiozen\src\qmapshack\CHelp.cpp(46): note: Es gibt keinen Kontext, in dem diese Konvertierung möglich ist
3>CMainWindow.cpp
~~~

Compilation with Qt 5.13.2 was successful.

**Comment by wthaem** [comment]
I followed my own recommendations under Ubuntu. Result: What looks nice under Windows is somehow broken under Ubuntu. Even the assistant doesn't display the help correctly. Maybe a Qt version conflict: Windows and Ubuntu Qt versions are different, Ubuntu version is much older. Will work on this.

**Comment by wthaem** [comment]
* I updated QMSHelp.* on the Wiki playground. This update is a QMS only help version. The QMT part isn't there anymore. Using this package with QMS on Windows looks ok. And it looks ok with assistant under Ubuntu.
* I mentioned earlier a `cache` folder (created by assistant only (?) in my home directory where I have my QMSHelp.* files). I had this folder under Ubuntu, too. And it happened to me with the Ubuntu assistant that the cached QMSHelp was displayed - it was the complete version with the QMS and QMT parts! And this instead of the version I had given on the command line. Strange. Conclusion: while testing delete this cache folder.

**Comment by wthaem** [comment]
There seem to be 2 reasons for the strange rendering of the first version of the Qt help: 
1. A badly configured help project file (in the filter part).
1. Rendering (including filtering of the content) depends on the help browser (assistant - QMS, Windows - Linux, different Qt versions).
I guess, that in some cases filtering took place and only part of the help information (the QMT part) passed the filter, in other cases the filter wasn't applied to the content and the content was completely shown.
Would it be possible to add tabs for the content tree, the index and for search to the QMS help browser? Then more checks could be done.

**Comment by wthaem** [comment]
There has been an important change in Qt help filter handling in Qt assistant version 5.14.0. While assistant version 5.13.x works as described in the Qt documentation, assistant version 5.14.0 has not yet documented changes in filter handling (compare the windows shown when going to `Edit > Preferences > Filters` in the assistant). Only some images have been updated in the Qt docu, not the description. When using Qt help in QMS the help engine version is the Qt version given to QMS when compiling it.

Thus, the phrase `Qt5.11 or later` in the QMS compile instructions for Windows certainly needs in the future some additional clarification.

_Recommendations for using Qt help with QMS/QMT:_ 
* Use 2 separate help packages for QMS and QMT.
* Don't define any filters in the package qhp configuration scripts.

**Comment by kiozen** [comment]
Help for QMS and QMT are two different things anyway.

The help dialog has to be improved for sure. But it will take some time as my resources are limited.

**Comment by kiozen** [comment]
Updated help dialog.

**Comment by wthaem** [comment]
@kiozen: The index works at first go (tested with Windows). I could imagine that this will please one or the other user who doesn't have to go to the internet for getting help info.

With the food comes the appetite (from German), i.e. there are more suggestions:

* (already mentioned and probably already on your list): Similar to the index tab, the search tab for full text search would be great.

* One is tempted to think of context-sensitive help. Here I have no idea how many changes this requires in the code. There are x places/situations where one can imagine this help (and where there is already a help text available in the Wiki). Is it possible to think in this direction or is it better not to? In the help package this is already prepared (for the newer help packages) by the fact that every term in the index can also be used as an "id" to call this position directly from the code. Additional id's can be easily defined. A list of the form "id: location_on_a_wiki_page" is all one would need additionally when creating the help package.

**Comment by wthaem** [comment]
My workflow for generating QtHelp:

* __Preparation/Installation:__
    * _Only once:_ Check resp. adjust Python and Qt installations on server (I don't know, if it is the server itself or kiozen's machine, doesn't matter) - should be done by the one having direct access to the server
    * _Repeat after script updates (seen from client side):_ Add resp. update Makefile, some Python and some Qt help project files to the local copy of the QMS Wiki and push the Wiki to the server (that is the only way for me to get easy and quick access to the server).
    * Adjust a few path settings in Makefile (documented in the Makefile: Python, qhelpgenerator) on the server and check the configuration settings using the Makefile itself. 
* __Generate Qt help:__ 
    * Run Makefile. Currently, the Makefile is there, where the MD Wiki files are and the result of running the Makefile are the QMS and QMT help packages located in the same directory. The Makefile run converts the Wiki pages (.md and some other files) to .html and packs the .html files into the help package.
    
I checked this workflow locally with Windows10 (using msys for running make) and with my VMWare Ubuntu 18.04. In both cases I can generate the help packages but certainly someone will find still hidden errors in the future. One possible reason for errors: Someone editing the Wiki violates the assumptions made about the Wiki.

@kiozen: Does this workflow match your requirements? How should it be improved?

**Comment by kiozen** [comment]
I tried the search stuff, but it did not produce any result. Probably I did something wrong. Maybe the files need extra stuff for a search.

Context sensitive help: Would be nice, but out of the scope of my time. But you can do it if you want. Shouldn't require that much knowledge of coding/c++. :wink: 

Workflow: I have to test it in real life. Does the Wiki page/Makefile document the needed sequence of commands?

**Comment by wthaem** [comment]
> I tried the search stuff, but it did not produce any result. Probably I did something wrong. Maybe the files need extra stuff for a search.

What does the last phrase mean? Which files need extra stuff? Using assistant the full text search works immediately.

> Context sensitive help: Would be nice, but out of the scope of my time. But you can do it if you want. Shouldn't require that much knowledge of coding/c++. 😉

Could you provide just one working example to have a template for including step-by-step more context-sensitive help? Then I could try to do some work in this direction.

> Workflow: I have to test it in real life. Does the Wiki page/Makefile document the needed sequence of commands?

I hope that I had included enough documentation in the scripts but it will take still a few days before I can make the scripts public. The Makefile is already in a quite could shape. Maybe, you can have a look at the attached file.

[Makefile.zip](https://github.com/Maproom/qmapshack/files/4271635/Makefile.zip)

**Comment by wthaem** [comment]
@kiozen: What do you think about including a call to git into the Makefile? This would make it easier to compose some file lists needed in the Makefile - but it is not a must, just a refinement.

**Comment by kiozen** [comment]
Should be ok.

**Comment by kiozen** [comment]
added full text search. But it's not working. Don't know why.

**Comment by wthaem** [comment]
Same result here. At least the tab is there.

From my naive point of view: Assuming that everything related to the search is in CHelpSearch.cpp this code seems to be quite short compared to the description found in https://doc-snapshots.qt.io/qt5-5.12/qhelpsearchengine.html. Should there be done more to show the list of locations where the search text was found?

**Comment by kiozen** [comment]
Fixed it.

**Comment by wthaem** [comment]
@kiozen:

* All files used for generating Qt help are now in the Wiki. Could you, please, test the help file generation on your machine? Does the workflow fit into your workflow for the preparation of a release?
* Start with reading the comments on top of the file https://github.com/Maproom/qmapshack/wiki/Makefile.make.
* Are the comments understandable and did you succeed to configure and run the Makefile? 
* Did you get QMSHelp.qch and QMSHelp.qhc (and a few other files!) in the root directory (there, where Makefile.make is stored)? Does the help look nice in the Qt help browser? Do links work correctly?
* Use the `clean` target to remove the intermediate .html files from the local wiki copy.

**Comment by wthaem** [comment]
@kiozen:

* All files used for generating Qt help are now in the Wiki. Could you, please, test the help file generation on your machine? Does the workflow fit into your workflow for the preparation of a release?
* Start with reading the comments on top of the file https://github.com/Maproom/qmapshack/wiki/Makefile.make.
* Are the comments understandable and did you succeed to configure and run the Makefile? 
* Did you get QMSHelp.qch and QMSHelp.qhc (and a few other files!) in the root directory (there, where Makefile.make is stored)? Does the help look nice in the Qt help browser? Do links work correctly?
* Use the `clean` target to remove the intermediate .html files from the local wiki copy.

**Comment by kiozen** [comment]
First problem :) On my SuSE system Qt4 and Qt5 are installed in parallel. SuSE solves this by appending the Qt helper binaries with `-qt5`. I solved this for now by:

```
QT5_SUFFIX ::= -qt5

# Example of OS-specific setup:
ifeq ($(OS),Windows_NT)
    QTDIR ::= d:/Qt/5.12.3/5.14.0/msvc2017_64/bin/
    PYTHON ::= python37
    HELPGENERATOR ::= $(QTDIR)qhelpgenerator${QT5_SUFFIX}
else
    QTDIR ::= 
    PYTHON ::= python3

    # qcollectiongenerator is flagged "deprecated" in newer Qt5 versions!
    # "qhelpgenerator" might not be available for older Qt5 versions!
    HELPGENERATOR ::= $(QTDIR)qcollectiongenerator${QT5_SUFFIX}    
endif
```
A more automated solution would be nicer.

**Comment by kiozen** [comment]
Looking good. I could create the files and they work. A few remarks:

* It would be nicer to have all the intermediate files in their own directory e.g. ./tmp
* It would be nicer to have all resulting files in their own directory e.g. ./doc_qms ./doc_qmt
* Right now I get more files than the two I need (`*.qhc`, `*.qch`). What are the others for? Can they go to the intermediate section?

Background: I would like to copy the content of the result folders 1:1 into the source path of the project.

As this needs quite a bunch of pre-requisites I am reluctant to move this as sub-repo into the main repo. This will make it harder for package maintainers. I would vote for copying the files into the source and commit them. Either on a regular base or right before each release. 

If you want you could do that as you have write permission to the repo. It would be ok to do this without a ticket and directly to dev. Same as it is done with the translations. But if you don't want it's ok, too. Then I will do it right before each release.

**Comment by wthaem** [comment]
Thanks for testing.I'm happy that you got the necessary files and that there seems to be a way to include this into your workflow and finally into the QMS distribution!
 
> First problem :) On my SuSE system Qt4 and Qt5 are installed in parallel. SuSE solves this by appending the Qt helper binaries with `-qt5`. I solved this for now by:
> QT5_SUFFIX ::= -qt5

I'll have a look at this and offer a solution.
 
> It would be nicer to have all the intermediate files in their own directory e.g. ./tmp

Could certainly be done. I just followed R.Woiteks script.

> It would be nicer to have all resulting files in their own directory e.g. ./doc_qms ./doc_qmt

Certainly possible.
> Right now I get more files than the two I need (`*.qhc`, `*.qch`). What are the others for? Can they go to the intermediate section?

You got the QMS*.* files - help for QMS and without any comment from my side 2 QMT*.* files - help for QMapTool. Should go to some doc* directory. 

> Background: I would like to copy the content of the result folders 1:1 into the source path of the project.

Will be possible if the help files are in some ./doc_qms resp. ./doc_qmt directory.
 
> As this needs quite a bunch of pre-requisites I am reluctant to move this as sub-repo into the main repo. 

Meaning is not completely clear to me. 

For the moment being I would prefer to have the files used for generating the help package within the Wiki (there, where they are now.). Intermediate files and the final help files will be moved to separate subdirectories within the Wiki. If the help generation has reached a stable state, then the intermediate files can be removed immediately by the Makefile. So, their directory is empty at the end of the run.

BTW: there is still R. Woitok's more or less outdated variant of creating HTML files for the Wiki pages. This was introduced as some kind of an offline help. Offering the Qt offline help to the user can certainly be seen as a complete replacement of Rainers HTMLMake procedure. Thus, I would propose to remove Rainers HTMLMake files.

> I would vote for copying the files into the source and commit them. Either on a regular base or right before each release.

That is my understanding, too. There is certainly no need to update the Qt help files after each change of a Wiki file. But there is definitely such a need before publishing a release. Just to initialize this procedure, please copy the 2 QMSHelp.* files to a convenient location in the qmapshack code repo. During the QMS compilation/installation these files have to be moved to the right location (which most likely depends on the operating system). The other 2 files "QMTHelp.*" are the equivalents for QMT. Thus, they have to be handled in a similar way, if you decide to include this help into QMT, too. My first attempt was to have just one help package for QMS **and** QMT (the one with a lot of problems in the first test, but this variant now works!). What is your opinion: one help package for QMS and one for QMT or just one package which can be called from both applications and where on the top level of the TOC you get the 2 entries "QMS help" and "QMT help".

> If you want you could do that as you have write permission to the repo. It would be ok to do this without a ticket and directly to dev. Same as it is done with the translations. But if you don't want it's ok, too. Then I will do it right before each release.

Can be discussed later.

**Comment by kiozen** [comment]
I get this list of files:
```
QMSAllHelp.qhcp
QMSHelp.qhcp
QMSHelp.qhp
QMTHelp.qch
QMTHelp.qhc
QMTHelp.qhcp
```
The *.qch and *.qhc files are known. What are the others for? Intermediate files?

> As this needs quite a bunch of pre-requisites I am reluctant to move this as sub-repo into the main repo.

One idea would be to make the Wiki repo a subrepo of the source code. And to integrate the generation of the help files into the make process. However this requires the installation of Python including some packages. And maybe some other stuff. Anyway it will make the process to compile more complex for the user. Especially novice users have a hard time already to compile and this will not improve if there is another batch of constrains added. That's why I am reluctant to follow that road.

HTMLMake: I agree to remove old stuff that's not needed any more. 

And, yes this has to be done for QMT, too. I will adapt the code before I start the pull request.

**Comment by wthaem** [comment]
QMS vs. QMT: QMapShack vs. QMapTool

*.qhp: Input configuration file for qhelpgenerator. Describes the detailed content of the help package (content table, index, used files,..). Called from *.qhcp. Generated with QMSQtHelp.py using info from existing Wiki. Must be updated after changes in Wiki pages. Intermediate file for the whole make process. Pretty big file.

*.qhcp: Input configuration file for qhelpgenerator ("c" stands for "container"). Describes general features of help package (container). Has pointer to .qhp. Created manually. Doesn't need update when Wiki page changes. *No intermediate file!*

*.qch/*.qhc: output of qhelpgenerator: the final help. Both files are needed when opening the help.
*.qhc as a container can consist of more than one .qch (for example: all-in-one help for QMS + QMT: 3 files involved: QMSHelp.qch, QMTHelp.qch, QMSAllHelp.qhc. A file QMSAllHelp.qhcp would describe this).

> ... to integrate the generation of the help files into the make process. However this requires the installation of Python including some packages. And maybe some other stuff. ...

* Wiki as subrepo: Wiki has easier access (write) rights than source code repo. Is it possible to give write access for everybody to a subrepo?
* Qt help files: My proposal would be to include the .qhc/.qch files into the source code repo. This would be similar to the .qm language files in the source. You (kiozen) compile from time to time the .qm files and you (or someone in charge of this step) compiles from time to time the help files. If somebody is interesting to do it by himself, he can try it. See the warnings concerning the compilation of the language files which apply to the help compilation, too. With this approach we need only a few people knowing how to compile the help on their well-configured computers just by running a Makefile.

    What is more important to me and where I don't know an answer yet: How do the package maintainers include the help files in their packages? In other words: where to put the help files during the installation? Is there one location for all OS or does the location depend on the OS? For Windows this should be an easy addition to Helmut's installer.

**Comment by wthaem** [comment]
@Kiozen: The whole lot is now restructured with the following additional doc branch in the local Wiki copy (subdirectories created by Makefile):

~~~
root (*.qhcp, *.qhp) 
      |
      --- doc (some general files used) 
           |
           --- qms (QMSHelp.qhc/qch, Makefile.make)                                 
           |
           --- qmt (QMTHelp.qhc/qch)
           |
           --- html (.html)
                |
                --- css (CSS file)
                |
                --- QMapTool (.html for QMT)
                |
                --- Downloads (.html for Downloads/*.wmts, ...)
~~~

                                                                        
The configuration in the Makefile has changed slightly: The QTDIR parameter has been removed. Now, the filenames (possibly with path) for the qhelpgenerator/qcollectiongenerator and for the assistant must be set as parameters. My settings for Windows and Ubuntu are shown as examples in the Makefile.                                                             
                                                                        
*For QMS help:* copy subdirectory doc/qms (with 2 files) to the required location in the source code repo. `doc/QMSAllHelp.qhc` is the combined QMS&QMT help.

Pay attention to the fact that the paths/addresses in the help packages have changed! For QMS they are now 2 levels lower. Please compare the `QMSHelp.qhcp` file to see a valid address. At least in `CHelp.cpp` there must be an update.

*Remark:* The strange "doc/doc" in the addresses is the result of a composition of the virtual folder name (doc) defined in the .qhp file (taken from the Qt documentation example) and the name of the new Wiki subdirectory (also doc).

In the Ubuntu version a call to "about:blank" results in an error message. In the Windows version this works as expected. Don't understand why.

The files left in the root directory (qhcp, qhp and Makefile) make address/path handling in the scripts much easier. Moving them down in the directory hierarchy would lead to fairly irritating address references.                                                               
                                                                        
Could you please have a look at the texts used in the `doc/*about*.txt` files and update them, if necessary?

I made a change in the global `.gitignore` file of the Wiki to avoid uploading .html files to the server (I'm not sure, if this change arrives at the server!).

**Comment by kiozen** [comment]
After creating the help files I get the following git status:
```
Änderungen, die nicht zum Commit vorgemerkt sind:
  (benutzen Sie "git add <Datei>...", um die Änderungen zum Commit vorzumerken)
  (benutzen Sie "git restore <Datei>...", um die Änderungen im Arbeitsverzeichnis zu verwerfen)
        geändert:       Makefile.make
        geändert:       QMSHelp.qhp
        geändert:       doc/qms/QMSHelp.qch
        geändert:       doc/qms/QMSHelp.qhc
        geändert:       doc/qmt/QMTHelp.qch
        geändert:       doc/qmt/QMTHelp.qhc

Unversionierte Dateien:
  (benutzen Sie "git add <Datei>...", um die Änderungen zum Commit vorzumerken)
        doc/html/AdvDataHandling.html
        doc/html/AdvMapDetails.html
        doc/html/AdvProjActions.html
        doc/html/AdvProjects.html
        doc/html/AdvRealtime.html
        doc/html/AdvRoutes.html
        doc/html/AdvSetup.html
        doc/html/AdvToolbarAct.html
...
```

I edited `Makefile.make` to add the `-qt5` suffix. The rest of the changed files shouldn't be in the repository and should be ignored by git. Background: Right now I have to revert all changes to do another `git pull`. I would like to avoid that. It would be fine to have all generated files in `doc` and add it to `.gitignore`. 

As a matter of fact your git ignore did not make it into the commit. Did you do a 
```
git add .gitignore
git commit
git push
```

I will have a look at the URL change.

**Comment by kiozen** [comment]
The about files for QMT might want to reference the direct link into QMT docs. And same for QMS. No need to land on the Wiki main main page.

**Comment by kiozen** [comment]
Started a PR. Help is now available in QMT and QMS.

**Comment by kiozen** [comment]
@huirad This will need some adjustment in the upcoming release.

**Comment by wthaem** [comment]
@kiozen:
* Your git status is completely correct and can be explained as follows:
* I couldn't be absolutely sure that my scripts run on your machine without any problems. Therefore, I uploaded some of the files which the scripts should create and which are essential. Once I know that the files are ok on your machine this will be removed in the future (concerned files: .qhp, .qch, .qhc).
* Run `make -f Makefile.make clean` as the last step - this removes all .html and .qhp files (=intermediate files). This step can be included in the future as last part to a successful build run. This removes most of the entries in the above git list.
* After copying or better moving the generated .qhc and .qch files to their final destination outside the Wiki these files disappear from the git list (and the doc diretory), too.
* The only file in the doc branch that can't be ignored or cleaned is the .css file in `doc/html/css`. It is an (not yet mentioned) additional input file for the build process. Its location is optimal for the HTML files.
* I guess the suffix `qt5` wasn't added at the right location. If you set the parameter on the command line it should read `make HELPGENERATOR=qcollectiongenerator-qt5 ASSISTANT=assistant-qt5 -f Makefile.make ...` (I hope the names are correct). Then no change in the Makefile anymore. Or set the parameters in the script itself (but this changes of course the script): 

~~~
    HELPGENERATOR ::= qcollectiongenerator-qt5
    ASSISTANT     ::= assistant-qt5
~~~

This explains the entries in the list of git messages.

* All generated files with the exception of the .qhp ones are somewhere in the doc directory. The .qhp files are temporarily in the root directory and are removed with the above mentioned clean step.
* Hopefully, the `.gitignore` issue will be solved soon.
 
> The about files for QMT might want to reference the direct link into QMT docs. And same for QMS. No need to land on the Wiki main page.

Up to now, I see the about message only when using the assistant, that is, I'm already in the help package. Therefore,  a link in the about message window should go to somewhere in the outside world or it should be dropped at all. And I considered the Wiki as outside world. Maybe, I didn't choose the optimal page as a reference.

**Comment by kiozen** [comment]
`.gitignore` and clean works perfectly, now!

**Comment by kiozen** [comment]
@wthaem have you tested the PR? I would like to merge it if everything is fine.

**Comment by wthaem** [comment]
Yes, using commit 18ffabe on Windows.

I had to copy the 4 QM?Help.q* files from their source locations to the usual Windows directory msvc_64/Files/doc/HTML (this is the easy additional installation step for Windows). Then, everything with F1 help for QMS and QMT was ok.

I didn't check Ubuntu yet.

**Comment by kiozen** [comment]
thx for testing
